micrometer-core/src/main/java/io/micrometer/core/aop/TimedAspect.java

```diff
@@ -60,16 +61,13 @@ public TimedAspect(MeterRegistry registry, Function<ProceedingJoinPoint, Iterabl
     public Object timedMethod(ProceedingJoinPoint pjp) throws Throwable {
         Method method = ((MethodSignature) pjp.getSignature()).getMethod();
         Timed timed = method.getAnnotation(Timed.class);
-
-        if (timed.value().isEmpty()) {
-            return pjp.proceed();
-        }
+        final String metricName = timed.value().isEmpty() ? DEFAULT_METRIC_NAME : timed.value();
```

Comment:
Why not build it from the package/class/method name? With using a default, it will keep the cardinality down, but I'm wondering how useful it will be to the user

Reply:
I think this speaks to the intended use, which is why I hesitated at a default before. If you are timing a specific code path, then plotting the name alone (exploded by a tag that represents the package/class/method) provides a cumulative aggregate. This is similar to what we did in spring-integration:

![image](https://user-images.githubusercontent.com/1697736/37466835-bb690b32-282c-11e8-8dd1-65ba56b75438.png)

If they all had different names, you'd have to construct this chart manually.

Keep in mind, the number of distinct time series is the same if you vary the name or a tag.

Reply:
"Why not build it from the package/class/method name? With using a default, it will keep the cardinality down, but I'm wondering how useful it will be to the user": Because then it is close to impossible to template in grafana, with a common metric name it can be a dashboard, with templated dropdowns for class/method you want via tags.